index.html

```diff
@@ -13219,11 +13219,14 @@ <h2>Excluding Elements from the Accessibility Tree</h2>
 	</section>
 	<section id="tree_inclusion">
 	  <h2>Including Elements in the Accessibility Tree</h2>
-          <p>If not already excluded from the accessibility tree per the rules above in <a href="#tree_exclusion">Excluding Elements in the Accessibility Tree</a>, user agents MUST provide an <a>accessible object</a> in the <a>accessibility tree</a> for <abbr title="Document Object Model">DOM</abbr> <a>elements</a> that meet any of the following criteria:</p>
+          <p>If not excluded from the accessibility tree per the rules above in <a href="#tree_exclusion">Excluding Elements in the Accessibility Tree</a>, user agents MUST provide an <a>accessible object</a> in the <a>accessibility tree</a> for <abbr title="Document Object Model">DOM</abbr> <a>elements</a> that meet any of the following criteria:</p>
           <ul>
-            <li>Text nodes</li>
-            <li>Elements that may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>.</li>
-            <li>Elements that are focusable, or have an ID <a>attribute</a> and an ancestor with the <pref>aria-activedescendant</pref> attribute that matches the implicit or explicit <a>semantics</a> of the required context role. In either case, the element may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
+            <li>Elements that are not <a>hidden</a> and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>:
+              <ul>
+                <li>Elements that are <a>focusable</a> even if the element or one of its ancestor elements has its <sref>aria-hidden</sref> attribute set to <code>true</code>.</li>
+                <li>Elements that have an ID <a>attribute</a>, and have the role and context of a valid target of the <pref>aria-activedescendant</pref> property.</li>
```

Comment:
CC @alice. 

@carmacleod I think this one as written requires all potential activedescendants to be exposed to the accessibility tree at all times. Perhaps:

> Elements that are a valid target of <pref>aria-activedescendant</pref>.

Note:  I specifically did not call refer to attributes (`id` or `aria-*`) because element reflection—when complete—would likely make the attribute-focused prose obsolete.

Reply:
@cookiecrook I love the brevity of this, and agree that we've already defined "a valid target of aria-activedescendant" in other places in the spec. (e.g. [Managing Focus](https://w3c.github.io/aria/#managingfocus), [aria-activedescendant](https://w3c.github.io/aria/#aria-activedescendant), etc).

Just to double-check, you're saying that potential activedescendants don't all have to be exposed to the accessibility tree at all times... so will user agents always be able to tell when to fire appropriate events?

Good point about `id`, but - yikes! I don't want to look at how many other places in the spec will need rewriting when element reflection is done... I believe we use `id` or `IDREF` all over the place (including in supporting docs/specs like APG, ACCNAME, etc).

Cross that bridge when we come to it? (Don't get me wrong - I'm looking forward to element reflection, just not looking forward to rewriting the specs... 😄 )

Reply:
> you're saying that potential activedescendants don't all have to be exposed to the accessibility tree at all times

As an example, most of the activedescendants in this example are hidden at most times. Even when the field is focused. For example, type "te" and all US states but Tennessee and Texas are hidden from the tree.

https://webkit.org/blog-files/aria1.0/combobox_with_live_region_status.html

Reply:
> As an example, most of the activedescendants in this example are hidden at most times.

I don't see `aria-activedescendant` in that example because it's using "owned descendants", but it's similar enough that I believe you. 😄 

I'm going with your suggestion:
> Elements that are a valid target of aria-activedescendant.

Reply:
> I don't see aria-activedescendant in that example because it's using "owned descendants", but it's similar enough that I believe you. 😄

Ack! Sorry for pointing you to the wrong example, but I'm thankful the point came across anyway.